metals/src/main/scala/scala/meta/internal/implementation/ImplementationProvider.scala

```diff
@@ -288,7 +289,7 @@ final class ImplementationProvider(
         classContext,
         source.toNIO,
       )
-      file <- locationsByFile.keySet.toArray.par
+      file <- locationsByFile.keySet.toArray
```

Comment:
@alexarchambault looks like parallel collections started hanging due to RejectedExecutionException 

I am just checking it right now. I used it previously to now have to rebuild the entire for comprehension, which is already complex. I am not sure yet how to fix it correctly.

Reply:
Did somehow including ScalaCli cause the pararel collections to start failing? The futures seems to get rejected and never finished.

Reply:
Weirdly enough JdkVersionsSuite is now failing:
```
[error] ==> X tests.JdkVersionSuite.jdk-shell-version  10.011s java.util.concurrent.TimeoutException: Future timed out after [10 seconds]
[error]     at scala.concurrent.impl.Promise$DefaultPromise.tryAwait0(Promise.scala:248)
[error]     at scala.concurrent.impl.Promise$DefaultPromise.result(Promise.scala:261)
[error]     at scala.concurrent.Await$.$anonfun$result$1(package.scala:201)
[error]     at scala.concurrent.BlockContext$DefaultBlockContext$.blockOn(BlockContext.scala:62)
[error]     at scala.concurrent.Await$.result(package.scala:124)
[error]     at scala.meta.internal.builds.ShellRunner$.runSync(ShellRunner.scala:165)
[error]     at scala.meta.internal.metals.JdkVersion$.fromShell(JavaInteractiveSemanticdb.scala:247)
[error]     at tests.JdkVersionSuite.$anonfun$new$1(JdkVersionSuite.scala:12)
```
looks like anything that uses Await is failing :thinking:

Reply:
It seems it was using a global execution context. I removed that, let's see if it helps. Though I am still curious, why would that cause these issues.